### ISSUE TITLE
draft a scheme for deciding agreement of statistical type with distributions in some metamodel

### ISSUE THREAD

**Issue by riastradh-probcomp** [probcomp/bayeslite#460]
For example, it might reject attempts to model a categorical variable with a normal distribution.  This issue will be satisfied not when we have a clear, coherent design implemented that we will stand behind for the rest of time, but when we have a draft of an idea to experiment with in practice to play around with it.  The concept of 'statistical type' remains fuzzy.

The best candidate summary I have come up with so far for the definition of 'statistical type' is 'the topology of a support of a random variable'.  The 'topology' part lets us meaningfully distinguish numerical from cyclic, for example, both of which are supported on the entire real line, but with entirely different topologies.  We also want to distinguish, e.g., ordered counts from unordered names.

The statistical type determines what operations are meaningful on the values that might appear for it, such as 'computing the logarithm' (not meaningful if values may be negative) or 'drawing a histogram sorted by height' (probably not what you want if values have an intrinsic ordering) or 'drawing a scatter plot' (not meaningful for pairs of variables that are unordered names).

**Comment by vkmvkmvkmvkm** [comment]
Feras is (I believe) on this as part of his paper revisions, with the
notion of "statistical type" coinciding with "canonical base measure that
must dominate the measure associated with any distribution chosen during
modeling".

**Comment by fsaad** [comment]
Yes, In addition to the base measure there are also the qualitative invariants that define the statistical type more completely -- for instance a count variable is very different than an open set categorical (both taking values on N, the former numeric and ordered, the latter symbolic and unordered) and that information needs to be encoded besides the support/base measure.
